### PR TITLE
Simplify HTTP basic auth configuration

### PR DESCRIPTION
**Pull request by zaneb**
Make HTTP basic auth configuration simpler, more flexible, and a better fit and more secure when used with standard k8s deployment techniques.

* Allow basic_auth to be configured independently on client and server
  interfaces, based on the presence of the required configuration data,
  rather than using a single global USE_HTTP_BASIC environment variable.
* Expect the server credentials to be passed in the form of an
  HTTP_BASIC_HTPASSWD environment variable containing both the username
  and the *hash* of the password, in the htpasswd format. This is more
  secure, as it allows the container not to hold a copy of the password
  when it doesn't need it purely for authenticating connections.
* Expect client credentials to be passed in the form of a file named
  /basic_auth/ironic/auth-config, formatted as an ini config file
  setting the appropriate options (for basic auth, this is
  auth_strategy=http_basic, and the username and password options; however
  this mechanism should work unchanged for other auth strategies). This is
  more secure because in k8s the password is never passed as an
  environment variable nor written to disk, but remains in a tmpfs
  filesystem.

### CODE REVIEW

**Comment by zaneb** [comment]
Equivalent change in ironic: metal3-io/ironic-image#182
/cc @iurygregory

**Review by juliakreger — APPROVED** [review]
LGTM

File: runironic-inspector.sh [hunk 1]
```diff
@@ -16,15 +11,20 @@ cp $CONFIG $CONFIG.orig
...
+    crudini --set $CONFIG DEFAULT http_basic_auth_user_file "${HTPASSWD_FILE}"
 fi
 
-exec /usr/bin/ironic-inspector --config-file /etc/ironic-inspector/inspector-dist.conf \
-	--config-file /etc/ironic-inspector/inspector.conf
+# Configure auth for ironic client
+CONFIG_OPTIONS="--config-file /etc/ironic-inspector/inspector-dist.conf --config-file ${CONFIG}"
+auth_config_file="/auth/ironic/auth-config"
+if [ -f ${auth_config_file} ]; then
+    CONFIG_OPTIONS+=" --config_file ${auth_config_file}"
```
Comment (dhellmann) [inline runironic-inspector.sh]: Should that be `--config-file` instead of `--config_file`?

Comment (zaneb) [reply runironic-inspector.sh]: yep, thanks

**Review by dhellmann — APPROVED** [review]
/lgtm

**Comment by iurygregory** [comment]
lgtm

**Review by iurygregory — APPROVED** [review]

**Pull request merged**